syntax = "proto3";

option go_package = ".;pb";

import "google/protobuf/empty.proto";

message CreateBootstrapTokenRequest {
  string expires_at = 1;
}

message CreateBootstrapTokenResponse {
  string token = 1;
}

message Host {
  string name = 1;
  string address = 2;
}

message ListHostsRequest {}

message ListHostsResponse {
  repeated Host hosts = 1;
}

message RegisterHostRequest {
  Host host = 1;
  string token = 2;
}

message HeartbeatRequest {
  Host host = 1;
}

message HeartbeatResponse {
}

service HostService {
  rpc CreateBootstrapToken(CreateBootstrapTokenRequest) returns (CreateBootstrapTokenResponse) {}
  rpc GetHost(Host) returns (Host) {}
  rpc ListHosts(ListHostsRequest) returns (ListHostsResponse) {}
  rpc Register(RegisterHostRequest) returns (Host) {}
  rpc Deregister(Host) returns (google.protobuf.Empty) {}

  rpc Heartbeat(HeartbeatRequest) returns (HeartbeatResponse) {}
}
